queue_job/job.py

```diff
@@ -233,22 +241,38 @@ def load(cls, env, job_uuid):
         job_.max_retries = stored.max_retries
         if stored.company_id:
             job_.company_id = stored.company_id.id
+        job_.identity_key = stored.identity_key
         return job_
 
+    @classmethod
+    def exist_duplicate_job(cls, env, identity_key):
+        """Check if a job to be executed with the same key exists."""
+        jobs = env['queue.job'].search(
+            [('identity_key', '=', identity_key),
+             ('state', 'in', [PENDING, ENQUEUED, STARTED])],
```

Comment:
tl; dr: Please remove `STARTED` from the search

----

On a second though, I'm wondering if we want all states. If the process is quite long. Would we have an issue of not creating a new job is one is already `STARTED`?

What I have in mind is for an instance:
- an update on record X creates a job to export
- later the job A starts
- while job A is in started state, a second update is made on record X
- it doesn't trigger a new export job

Thus you will have a mismatch between local recordA and it's exported values.
And you won't have any job to update it. And might never have another update on that record.

This could happen if the current job is running on outdated values.

I see 2 options to avoid this.

1. destroy/create or restart the existing job

When next job should do exactly the same we want to do a retry with udpated data.
(as far as I know an ongoing job won't be killed by changing the state so it is more a do again than a restart)

Or could we imagine interrupting the ongoing job to really restart it? But we would have to add some mechanisms to interrupt nicely.

2. don't consider `STARTED` as duplicate

This would clearly mean sometime doing the same thing and create a new job. But it would close the breach.
Having 2 jobs would show more clearly that you did 2 calls, even if the first was useless.

Reply:
I think @yvaucher is correct and a started job should not be considered as a duplicate.

Reply:
> destroy/create or restart the existing job

No please :dagger: 

> don't consider STARTED as duplicate

Yes